Jenkinsfile

```diff
@@ -6,44 +6,45 @@ def buildSuccess = false
 def rosContainer
 try {
   node('android') {
-    // Allocate a custom workspace to avoid having % in the path (it breaks ld)
-    ws('/tmp/realm-java') {
-      stage('SCM') {
-        checkout([
-          $class: 'GitSCM',
-          branches: scm.branches,
-          gitTool: 'native git',
-          extensions: scm.extensions + [
-            [$class: 'CleanCheckout'],
-            [$class: 'SubmoduleOption', recursiveSubmodules: true]
-          ],
-          userRemoteConfigs: scm.userRemoteConfigs
-        ])
-      }
-
-      def buildEnv
-      def rosEnv
-      stage('Docker build') {
-        // Docker image for build
-        buildEnv = docker.build 'realm-java:snapshot'
-        // Docker image for testing Realm Object Server
-        def dependProperties = readProperties file: 'dependencies.list'
-        def rosDeVersion = dependProperties["REALM_OBJECT_SERVER_DE_VERSION"]
-        rosEnv = docker.build 'ros:snapshot', "--build-arg ROS_DE_VERSION=${rosDeVersion} tools/sync_test_server"
-      }
-
-      rosContainer = rosEnv.run('-v /tmp=/tmp/.ros')
-
-      try {
+    timeout(time: 1, unit: 'HOURS') {
```

Comment:
Perhaps set this to 2 hours? I think a full clean build comes pretty close to this, especially if running on a slow machine / device.